### PR TITLE
Build docker image

### PR DESCRIPTION
**Pull request by clee2000**
Downloading pip dependencies takes 2-4 min, but downloading the new docker image takes +1min from the pytorch one
Not sure how downloading all dependencies to a new image takes in comparison to pulling docker image

Docker image scripts are copied from pytorch, I tried to remove stuff but there's definitely a lot left over

### CODE REVIEW

File: .ci/docker/common/install_conda.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  # Prevent conda from updating to 4.14.0, which causes docker build failures
+  # See https://hud.pytorch.org/pytorch/pytorch/commit/754d7f05b6841e555cea5a4b2c505dd9e0baec1d
+  # Uncomment the below when resolved to track the latest conda update
+  # as_jenkins conda update -y -n base conda
+
+  # Install correct Python version
+  as_jenkins conda create -n py_$ANACONDA_PYTHON_VERSION -y python="$ANACONDA_PYTHON_VERSION"
+
+  # Install PyTorch conda deps, as per https://github.com/pytorch/pytorch README
+  CONDA_COMMON_DEPS="astunparse pyyaml mkl=2021.4.0 mkl-include=2021.4.0 setuptools"
```
Comment (malfet) [inline .ci/docker/common/install_conda.sh:43]: Why do we need astunpar, pyyaml, mkl etc?

File: .ci/docker/common/install_conda.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  # HACK HACK HACK
+  # gcc-9 for ubuntu-18.04 from http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu
+  # Pulls llibstdc++6 13.1.0-8ubuntu1~18.04 which is too new for conda
+  # So remove libstdc++6.so.3.29 installed by https://anaconda.org/anaconda/libstdcxx-ng/files?version=11.2.0
+  # Same is true for gcc-12 from Ubuntu-22.04
+  if grep -e [12][82].04.[623] /etc/issue >/dev/null; then
+    rm /opt/conda/envs/py_$ANACONDA_PYTHON_VERSION/lib/libstdc++.so.6
+  fi
+
```
Comment (malfet) [inline .ci/docker/common/install_conda.sh:71]: We don't need this hack, as we don't do builds in tutorial builds

File: .ci/docker/common/install_cudnn.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        exit 1
+    fi
+
+    tar xf ${CUDNN_NAME}.tar.xz
+    cp -a ${CUDNN_NAME}/include/* /usr/local/cuda/include/
+    cp -a ${CUDNN_NAME}/lib/* /usr/local/cuda/lib64/
+    popd
+    rm -rf tmp_cudnn
+    ldconfig
+fi
```
Comment (malfet) [inline .ci/docker/common/install_cudnn.sh:24]: We don't need installing CUDNN at all
```suggestion

```

File: .ci/docker/common/install_user.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+# Allow writing to /usr/local (for make install)
+chown jenkins:jenkins /usr/local
+
+# Allow sudo
+# TODO: Maybe we shouldn't
+echo 'jenkins ALL=(ALL) NOPASSWD:ALL' > /etc/sudoers.d/jenkins
+
+# Test that sudo works
+sudo -u jenkins sudo -v
```
Comment (malfet) [inline .ci/docker/common/install_user.sh:26]: Why do we need this one?

File: .ci/docker/build.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+set -exu
+
+IMAGE_NAME="$1"
+shift
+
+export CUDA_VERSION="12.1.1"
+export UBUNTU_VERSION="20.04"
+
+export BASE_IMAGE="nvidia/cuda:${CUDA_VERSION}-devel-ubuntu${UBUNTU_VERSION}"
```
Comment (malfet) [inline .ci/docker/build.sh:16]: NVIDIA image contains tons of dev tools which are not really needed to build/test tutorials, are they?
```suggestion
export BASE_IMAGE="ubuntu:${UBUNTU_VERSION}"
```

File: .ci/docker/common/install_user.sh (hunk 4 above)
Comment (clee2000) [reply .ci/docker/common/install_user.sh]: Removed, also removed references to jenkins but didn't have chown the repository

File: .ci/docker/common/install_base.sh [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    build-essential \
+    ca-certificates \
+    cmake=3.16* \
+    curl \
+    git \
+    wget \
+    sudo \
+    vim \
+    jq \
+    vim \
```
Comment (malfet) [inline .ci/docker/common/install_base.sh:20]: I really like `vim`, but installing it once is sufficient
```suggestion
```

**Review by malfet — APPROVED** [review]
LGTM, thank you for all the updates

**Pull request merged**